src/key.rs

```diff
@@ -28,13 +28,22 @@ use Verification;
 use constants;
 use ffi::{self, CPtr};
 
+#[cfg(feature = "serde")]
+use serde::ser::SerializeTuple;
+
 #[cfg(feature = "global-context")]
 use {Message, ecdsa, SECP256K1};
 #[cfg(all(feature  = "global-context", feature = "rand-std"))]
 use schnorr;
 
 /// Secret 256-bit key used as `x` in an ECDSA signature.
 ///
+/// # Serde support
+///
+/// Implements de/serialization with the `serde` feature enabled. Serialized data is fixed width (32
+/// bytes) if you use `bincode` to do serialization, other binary formats may add metadata to the
+/// serialized data (e.g. cbor and tycho).
+///
```

Comment:
I think it'd be very useful to explicitly say we treat it as tuple of 32 `u8`s for non-human formats. I think it's a good idea to mention serialization formats but I would word it like this:

"This representation is optimal for e.g. bincode format. Some formats may be less optimal."

Reply:
While implementing this suggestion I realized that we do not differentiate between human-readable and non-humand-readable in secp as we do in bitcoin. Is this by design or an omission?

Reply:
This also means that the docs are now slightly misleading because they imply that we do something different for human-readable formats.

> Implements de/serialization with the serde feature enabled. We treat the byte value as a tuple of 33 u8s for non-human-readable formats. This representation is optimal for for some formats (e.g. [bincode](https://docs.rs/bincode)) however other formats may be less optimal (e.g. [cbor](https://docs.rs/cbor)).

Reply:
Damn, I realized it may sound like we don't implement human-readable format. So should say we do it as hex.

Reply:
I documented the hex string for human-readable formats.